message-queue-lib/test/cmr/message_queue/test/queue/sqs.clj

```diff
@@ -39,3 +45,62 @@
 
       "OPS without prefix already applied"
       "ops" "abc123" "gsfc-eosdis-cmr-ops-abc123")))
+
+(defn- queue-request-proxy
+  "Creates a proxy for GetQueueUrlRequest that returns the given
+  string for calls to getQueueUrl"
+  [url-str]
+  (proxy [GetQueueUrlResult]
+         []
+         (getQueueUrl [] url-str)))
+                         
+(defn- client-proxy
+  "Creates a proxy for AWSSQSClient. The proxy returns a GetQueueUrlRequest
+  proxy for calls to getQueueUrl. For calls to receiveMessages it calls the
+  functions in receive-fns starting with the first function and then using 
+  the next function in the list for each subsequent call until the list of
+  receive functions is exhausted. At that point subsequent calls to 
+  receiveMessages will throw a Throwable to force the calling thread to
+  exit. This is done to prevent test worker threads from persisting after
+  tests complete."
+  [url-str receive-fn-index & receive-fns]
+  (proxy [AmazonSQSClient]
+         []
+         (getQueueUrl [_q-name] (queue-request-proxy url-str))
+         (receiveMessage [_req] (let [index (swap! receive-fn-index inc) 
+                                      rec-fn (if (< index (count receive-fns))
+                                                 (nth receive-fns index)
+                                                 #(throw (Throwable. "Exiting thread")))]
```

Comment:
What about instead of a general `Throwable` we define our own exception for this particular type of use? I'll update the code above with some thoughts on this ...